test.js

```diff
@@ -0,0 +1,313 @@
+  
+// function removeUnaryPlus(expression) {
+//   // Replace unary '+' operators with empty strings
+//   const regex = /(^|\()(\+)(?=\d)/g;
+//   return expression.replace(regex, '$1');
+// }
+
+
+// const expression = "+1+(+2)";
+// const result = removeUnaryPlus(expression);
+// console.log(result); // Output: "2 + 3 + 5"
+
+
+
+
+
+function isValidExpression(expression) {
+  const stack = [];
+  const openingBrackets = ['(', '[', '{'];
+  const closingBrackets = [')', ']', '}'];
+  
+  for (const char of expression) {
+      if (openingBrackets.includes(char)) {
+          stack.push(char);
+      } else if (closingBrackets.includes(char)) {
+          const lastOpeningBracket = stack.pop();
+          const correspondingOpeningBracket = openingBrackets[closingBrackets.indexOf(char)];
+          
+          if (lastOpeningBracket !== correspondingOpeningBracket) {
+              return false; // Mismatched brackets
+          }
+      }
+  }
+  
+  return stack.length === 0; // All brackets are properly balanced
+}
+
+
+
+function radiansToDegrees(radians) {
+  return parseFloat(radians) * (180 / Math.PI);
+}
+
+
+
+function isOperator(token) {
+  const operators = "+-*/";
+  return operators.includes(token);
+}
+function isNumber(token) {
+  return !isNaN(parseFloat(token)) && isFinite(token);
+}
+
+function changeExpression(tokens,dict)
+{
+let output=[];
+let temp='';
+let j;
+for(let i=0;i<tokens.length;i++ )
+{
+  temp='';
+  
+    if (tokens[i] === 'Math.sin' || tokens[i] === 'Math.cos' || tokens[i] === 'Math.tan') 
+    {
+      if(tokens[i]=='Math.tan' && tokens[i+2]=='90')
+      {
+        return 'undefined ';
+      }
+      output.push(tokens[i]);
+      j=parseInt(i)+2;
+      
+       
+      temp=radiansToDegrees(tokens[j]);
+      //temp=parseFloat(tokens[j]);
+      temp ="("+temp+")";
+      i=parseInt(j)+1;
+      
+    }
+    else if(i<=j)
+    {
+      continue;
+    }
+  else if (isOperator(tokens[i]))
+  {
+    temp=tokens[i];
+    //console.log(temp);
+
+  }
+  else if (isNumber(tokens[i]))
+  {
+    temp=tokens[i];
+    //console.log(temp);
+  }
+  else if(dict.hasOwnProperty(tokens[i]))
+  {
+    temp=dict[tokens[i]];
+    //console.log(temp);
+  }
+ 
+  output.push(temp);
+}
+
+ return output.join('');
+
+}
+
+
+
+let exp='tan(45)';
+let dict={'w':20};
+exp=exp.replace('sin','Math.sin');
+exp=exp.replace('cos','Math.cos');
+exp=exp.replace('tan','Math.tan');
+exp=exp.replace('e','Math.E');
+exp=exp.replace('pi','Math.PI');
+if(!isValidExpression(exp)){
+  console.log('not a valid expression ');
+  return;
+}
+const tokens = exp.match(/(?:sin|cos|tan|sqrt|\b[A-Za-z0-9.]{1,9}(?![\d.])\b|\^|\b[+\-*/()]\b|e|pi|((?:\d+(?:\.\d)?|\.\d+|[+\-*/()])))?/g);
+//console.log('Before changing' + tokens);
+let changedExpression=changeExpression(tokens,dict);
+if(changedExpression=='undfined')
+{
+  console.log('Undefined');
+}
+//console.log('After changing' + changedExpression);
```

Comment:
remove commented out code

Reply:
sorry for inconvenience, but my project's files are Calculator.html, script1.js , style.css.  Rest are extra files, I deleted all of them now.